### PR TITLE
Add nodeAffinity to csi daemonset

### PR DESCRIPTION
**Pull request by luhi-DT**
## Description

Ticket: https://dt-rnd.atlassian.net/browse/K8S-9882

In order to align operator components, we should add nodeAffinity to the CSI Deamonset, as Operator/webhook already have it.

## How can this be tested?

Rollout the CSI Driver on a kubernetes cluster -> check if nodeAffinity is set (GKE currently has another bug, as normal GKE is detected as autopilot, will fix that in another PR) so try to not use GKE for testing

### CODE REVIEW

File: config/helm/chart/default/templates/Common/csi/daemonset.yaml [hunk 1]
```diff
@@ -262,6 +262,23 @@ spec:
       {{- if .Values.csidriver.nodeSelector }}
       nodeSelector: {{- toYaml .Values.csidriver.nodeSelector | nindent 8 }}
       {{- end }}
+      affinity:
```
Comment (chrismuellner) [inline config/helm/chart/default/templates/Common/csi/daemonset.yaml:265]: could we move this to a template and reuse it for operator, webhook and csi driver? This way we'd only have to change something in one place when adding a new architecture or modifying `nodeAffinity`/`tolerations` for a different reason.

Comment (luhi-DT) [reply config/helm/chart/default/templates/Common/csi/daemonset.yaml]: done

File: config/helm/chart/default/templates/Common/csi/daemonset.yaml [hunk 2]
```diff
@@ -262,22 +262,12 @@ spec:
...
-        - key: kubernetes.io/arch
-          value: amd64
-          effect: NoSchedule
-        - key: kubernetes.io/arch
-          value: ppc64le
-          effect: NoSchedule
-        - key: kubernetes.io/arch
-          value: s390x
-          effect: NoSchedule
+        {{- include "dynatrace-operator.defaultTolerations" . | nindent 8 }}
```
[the remark this answers is not in the document]
Comment (luhi-DT) [reply config/helm/chart/default/templates/Common/csi/daemonset.yaml:270]: does it matter for tolerations? i guess not, if they are there they are there :D

File: config/helm/chart/default/templates/_platform.tpl [hunk 3]
```diff
@@ -52,3 +52,41 @@ Enforces that platform is set to a valid one
...
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
+      nodeSelectorTerms:
+        - matchExpressions:
+              {{- if ne (include "dynatrace-operator.platform" .) "gke-autopilot"}}
+            - key: kubernetes.io/arch
+              operator: In
+              values:
+                - amd64
```
[the remark this answers is not in the document]
Comment (luhi-DT) [reply config/helm/chart/default/templates/_platform.tpl:66]: we could do that, but i am somehow against it, for me the helm manifests should be kinda readable, putting logic in them which go over if's or some templates its unreadable how the manifest is generated

**Review by chrismuellner — DISMISSED** [review]

**Review by chrismuellner — APPROVED** [review]

**Pull request merged**